utility/types/message.go

```diff
@@ -1,53 +1,25 @@
 package types
 
 import (
-	"bytes"
 	"encoding/hex"
 	"log"
-	"net/url"
-	"strconv"
-	"strings"
 
 	"github.com/pokt-network/pocket/shared/codec"
 	coreTypes "github.com/pokt-network/pocket/shared/core/types"
 	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
-	"google.golang.org/protobuf/proto"
-)
-
-/*
-`message.go`` contains `ValidateBasic` and `SetSigner`` logic for all message types.
-
-`ValidateBasic` is a **stateless** validation check that should encapsulate all
-validations possible before even checking the state storage layer.
-*/
-
-// CLEANUP: Move these to a better shared location or inline the vars.
-const (
-	MillionInt       = 1000000
-	ZeroInt          = 0
-	HeightNotUsed    = int64(-1)
-	EmptyString      = ""
-	HttpsPrefix      = "https://"
-	HttpPrefix       = "http://"
-	Colon            = ":"
-	Period           = "."
-	InvalidURLPrefix = "the url must start with http:// or https://"
-	PortRequired     = "a port is required"
-	NonNumberPort    = "invalid port, cant convert to integer"
-	PortOutOfRange   = "invalid port, out of valid port range"
-	NoPeriod         = "must contain one '.'"
-	MaxPort          = 65535
 )
 
+// A message is a component of a transaction (excluding metadata such as the signature)
+// defining the action driving the state transition
 type Message interface {
-	proto.Message
+	codec.ICodec
```

Comment:
I don't like this TBH. Care to elaborate the reasoning behind the change here?

Reply:
Also added the following comment in the interface

```go
	// TECHDEBT: This is necessary for us to be able to apply some `proto` specific transformations
	//           on types conforming to this interface
	codec.CodecType
```